src/components/NavContainer.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import { Text, View } from 'react-native';
+import { NavigationContainer } from '@react-navigation/native';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import Ionicons from 'react-native-vector-icons/Ionicons';
+
+import HomeScreen from '../screens/HomeScreen';
+import TaskScreen from '../screens/TaskScreen';
+import BadgeScreen from '../screens/BadgeScreen';
+import ProfileScreen from '../screens/ProfileScreen';
+
+const Tab = createBottomTabNavigator();
+
+export default function NavFlow() {
+  return (
+    <NavigationContainer>
+      <Tab.Navigator
+        screenOptions={({ route }) => ({
+          tabBarIcon: ({ color, size }) => {
+            let iconName;
+
+            if (route.name === 'Home') {
```

Comment:
It might be better to use a switch statement here. 

More importantly, however, I think 'Home', 'Tasks', etc. should be stored in constants. Same for lines 40-43

Reply:
What might even be more interesting is having an object that maps route names to icons, so that this entire function becomes 1-2 lines. Something like:
```jsx
const NavRoutes = { // to be used in lines 40-43 as well
  HOME: 'Home',
  TASKS: 'Tasks',
  PROGRESS: 'Progress',
  ME: 'Me',
}
const NavRouteToIcon = {
  [NavRoutes.HOME]: 'ios-home',
  [NavRoutes.TASKS]: 'ios-list',
  [NavRoutes.PROGRESS]: 'ios-clipboard',
  [NavRoutes.ME]: 'ios-person'
}
// in function:
const iconName = NavRoutesToIcon[route.name];
return <Ionicons name={iconName} size={size} color={color} />;
```

Reply:
To add to Shravan's , we can map route to icon + component (lines 40-43) as well